oidc/config.go

```diff
@@ -114,6 +120,78 @@ func NewConfig(issuer string, clientID string, clientSecret ClientSecret, suppor
 	return c, nil
 }
 
+// Hash will produce a hash value for the Config, which is suitable to use for
+// comparing two configurations for equality.
+func (c *Config) Hash() (uint64, error) {
+	var h uint64
+	var err error
+
+	algs := make([]string, 0, len(c.SupportedSigningAlgs))
+	for _, a := range c.SupportedSigningAlgs {
+		algs = append(algs, string(a))
+	}
+
+	scopes := make([]string, 0, len(c.Scopes))
+	scopes = append(scopes, c.Scopes...)
+
+	audiences := make([]string, 0, len(c.Audiences))
+	audiences = append(audiences, c.Audiences...)
+
+	redirects := make([]string, 0, len(c.AllowedRedirectURLs))
+	redirects = append(redirects, c.AllowedRedirectURLs...)
+
+	sort.Strings(algs)
+	sort.Strings(scopes)
+	sort.Strings(audiences)
+	sort.Strings(redirects)
+
+	args := make([]string, 0, len(algs)+len(scopes)+len(audiences)+len(redirects)+4)
```

Comment:
I think the final append of the `NowFunc` value would make the capacity of `args` need a `+5` instead of `+4`? No functional difference, but I think the final append would end up allocating a new underlying array since it'd be over capacity.

Reply:
ty! great catch.  I've rearrage things a bit to make it easier to align the capcity with the number of args.